libvast/src/system/spawn.cpp

```diff
@@ -11,91 +11,142 @@
  * contained in the LICENSE file.                                             *
  ******************************************************************************/
 
+#include "vast/system/spawn.hpp"
+
 #include <iterator>
 
-#include <caf/all.hpp>
+#include <caf/actor.hpp>
+#include <caf/actor_cast.hpp>
+#include <caf/config_value.hpp>
+#include <caf/dictionary.hpp>
+#include <caf/expected.hpp>
+#include <caf/send.hpp>
+#include <caf/string_algorithms.hpp>
 
 #include "vast/config.hpp"
 
 #include "vast/concept/parseable/to.hpp"
 #include "vast/concept/parseable/vast/expression.hpp"
+#include "vast/concept/parseable/vast/schema.hpp"
 #include "vast/concept/printable/vast/expression.hpp"
+#include "vast/concept/printable/vast/schema.hpp"
 #include "vast/data.hpp"
 #include "vast/defaults.hpp"
+#include "vast/detail/make_io_stream.hpp"
 #include "vast/error.hpp"
 #include "vast/expression.hpp"
 #include "vast/expression_visitors.hpp"
+#include "vast/format/ascii.hpp"
+#include "vast/format/bgpdump.hpp"
+#include "vast/format/bro.hpp"
+#include "vast/format/csv.hpp"
+#include "vast/format/json.hpp"
+#include "vast/format/mrt.hpp"
+#include "vast/format/test.hpp"
 #include "vast/query_options.hpp"
-
-#include "vast/system/atoms.hpp"
+#include "vast/schema.hpp"
 #include "vast/system/archive.hpp"
+#include "vast/system/atoms.hpp"
+#include "vast/system/exporter.hpp"
 #include "vast/system/importer.hpp"
 #include "vast/system/index.hpp"
-#include "vast/system/exporter.hpp"
+#include "vast/system/node.hpp"
 #include "vast/system/profiler.hpp"
-#include "vast/system/spawn.hpp"
 #include "vast/system/replicated_store.hpp"
+#include "vast/system/sink.hpp"
+#include "vast/system/source.hpp"
+
+#ifdef VAST_HAVE_PCAP
+#include "vast/format/pcap.hpp"
+#endif // VAST_HAVE_PCAP
+
+// Save ourself repeated lines of code that either declare a local variable or
+// return with an error.
+#define UNBOX_VAR(var_name, expr)                                              \
```

Comment:
(Sigh...C++ needs monads!)

For a good style, we should always have a corresponding `undef` of locally defined macros. Even though it's an implementation file, the code may get refactored later on and land elsewhere.

Reply:
I don't follow your reasoning. This symbol is only defined for this translation unit. What would an `undef` at the end accomplish, when it vanishes regardless a couple of lines later?

Besides, it seems you are reviewing commit-by-commit. Nothing wrong with that, but the downside of doing this is that one has to double-check whether the code you see wasn't changed some commits later. This particular macro has since moved to a detail header and was renamed to `VAST_UNBOX_VAR` since it no longer is just at file scope.

Reply:
> I don't follow your reasoning. 

The reasoning is consistency and robustness against later refactoring. Even though the macro has the scope of a single TU, it might not have it in the future.